### ISSUE TITLE
Fixes needed for GUI

### ISSUE THREAD

**Issue by Tomivix** [Tomivix/hpa#6]
~~The registers/memory part of the GUI could use some labels on top. Some free space/line between those two sections to have clear distinction would also come in handy.
Counting on you @Aicedosh 
(Also change the font to some simpler, monospaced sans would be the best)~~

**Comment by Aicedosh** [comment]
Already made a pull request

**Comment by Tomivix** [comment]
~~Good job @Aicedosh ! It looks much better now. Will leave this issue open in case I get some new ideas.
Also can you force all the letters in Directives/Orders section become capital, regardless of user input?
(not sure if this is possible but it should make whole thing look more assembler-esque...)~~

**Comment by Aicedosh** [comment]
Done (one line changed)

**Comment by Tomivix** [comment]
~~Well thanks for that @Aicedosh . That should do it for today.
(also it seems like i have to improve my Swing skills...)~~

**Comment by Tomivix** [comment]
~~One small request @Aicedosh : please make the arrow in Memory/Registers section have some other color. Black arrow among black rectangles and black section borders is hardly distinguishable. Also good job with the line numbering system!~~

**Comment by Tomivix** [comment]
~~Since syntax highlighting seems to work just okay, I have another request for you @Aicedosh . Can you add two buttons to the GUI - one labeled "Compile/Build" and the second one labeled "Run/Execute" (you can come up with different, better labels)? I'd like to start working on the actual processing of the code...~~

**Comment by Aicedosh** [comment]
Okay, I'll try to add it together with registers and memory cells value edition as a next update

**Comment by Tomivix** [comment]
~~Okay, looking forward to it.~~

**Comment by Tomivix** [comment]
~~Tested your new version of GUI @Aicedosh and found a couple of issues:~~
- ~~Is it just me or the box of the 15th register doesn't completely fit inside the window?~~
- ~~Manual edition of the registers is great but if you try entering wrong value (like '2' for binary,'a' for decimal or 'g' for hex) the field still accepts it. Not only does it throw errors in the console but it also permanently breaks whole format conversion system. You should add some input restrictions.~~
- ~~Manually selecting the number every time you want to edit sucks. Make it automatically select all content of the field every time particular JTextField gets focused. Also make it auto focus and select all content of the 'Decimal' field rather than 'Binary' every time you open register edit dialog.~~

**Comment by Aicedosh** [comment]
Yup, I'm aware of those issues. I had to go out some time ago but I wanted to make a pull request before that. I'll fix them ASAP but probably not today

**Comment by Tomivix** [comment]
~~Okay, then I'm looking forward to your next update. Also thanks for adding the buttons, now I can finally start working on the actual processing of the code. BTW take a look at #16 and give your opinion there.~~

**Comment by mrwasp** [comment]
Looks quite good.

What is the ~~destination~~ _point_ of manual edition of registers? Debugging? Testing? Or just fun? :)

**Comment by Tomivix** [comment]
~~No idea @mrwasp . I don't see any real usage for it either, but @Aicedosh seemed pretty keen on adding it for some reason. Since it does no harm I think we should leave it for now. We can always turn it off later...~~

**Comment by Aicedosh** [comment]
Well, I thought we were talking about it @mrwasp  as A way to input data.

**Comment by Tomivix** [comment]
~~Looks like the GUI window will need to have its height set dynamically. Right now some other parts of the OS interface (e,g start bar) can overlay it, which combined with blocked resize makes it quite unpleasant.~~

**Comment by Aicedosh** [comment]
What's your screen resolution? I'm trying to figure out if scaling whole GUI based on screen resolution would work out

**Comment by Tomivix** [comment]
~~On this notebook here I have 1366x768 pixels but at home I have display with 1920x1080 pixels... 
By the way what do you need my screen resolution for if you agreed scaling should be dynamic?~~

**Comment by Aicedosh** [comment]
Just wasn't sure if resolution was the case. Anyways I added a pull request with temporary solution

**Comment by Aicedosh** [comment]
Here's the list you asked for @Tomivix 

```
* Color coding arrow depending on last order
* Highliting line with caret / currently executed order
* Put registers and cells on ScrollPane (fixing arrow coords needed) -- scaling window size solution
* Horizontal spacing between memory cells
* Pause program execution, add custom interval between steps
* Deactivate run button if code changed / make it run build if necessary
* Include orders in memory panel
* Update registers below each other arrow fix
* Save and load files
```

**Comment by Tomivix** [comment]
~~Many thanks for the list @Aicedosh . Too bad the issues you'll be working on right now aren't bolded out like before. Can you change that please? Then I will update the project accordingly.~~

**Comment by Aicedosh** [comment]
I'm not sure if it's worth to do it anyways. I already fixed the bug with updating register directly below each other and removed the arrow on jump execution. I also started putting memory cells and registers on scroll panes but got stuck. There are some weird size calculations going on and I've no idea why. I'll try to fix that today but I can't promise anything

**Comment by Tomivix** [comment]
You remember me talking about another bug this morning, right @Aicedosh? Turns out it doesn't work quite the way I described it to you. The real problem is the fact that the about/help dialog is independent from the main window (you can use it without previously closing of the dialog). I just clicked the button twice but closed only one if the dialogs. The other one kept showing in the background every time I opened edit register/memory cell box for some reason. So my request would be making it modular to the main window (or  whatever it was called) so that it stays on top and you can't proceed without dismissing it first.
